pywbemtools/pywbemcli/_utils.py

```diff
@@ -108,14 +109,50 @@ def _eq_name(name1, name2):
     return name1.lower() == name2.lower()
 
 
-def formatwarning(message, category, filename, lineno, line=None):
+def _formatwarning(message, category, filename, lineno, line=None):
     # pylint: disable=unused-argument
     """
-    Replacement for warnings.formatwarning() that is patched in.
+    Replacement for warnings.formatwarning() that replaces the standard
+    warning;.formatwarning for pywbemcli.
     """
     return "{}: {}\n".format(category.__name__, message)
 
 
+def pywbemcliwarn(*args, **kwargs):
+    """
+    Pywbemcli replacement for the warnings.warn function that substitutes our
+    own formatter.
+    """
+    # Import for each call
+    # pylint: disable=import-outside-toplevel, cyclic-import
+    from .pywbemcli import WARN_OPTION
+    if WARN_OPTION:
+        warnings.simplefilter('once')
+    # else: Leave warning control to the PYTHONWARNINGS env var.
+    # Delete the variable so it is imported with next call
+    del WARN_OPTION
```

Comment:
This `del` deletes the name `WARN_OPTION` from the local scope of the function, not from the pywbemcli module's namespace. I'm not sure I understand what the purpose is, since it will be gone anyway once the function returns?

Reply:
That one was strange but it forces the import to be redone.  The wonders of stackOverflow.  That causes the reimport to get the current value of the import.  Yes, it apparently works but I hated it the minute I used it.  However, once I determined better what the scope of the warnings filters was all this was not needed so removed all of the export WARN_OPTION, etc.